compiler/src/dotty/tools/dotc/core/Decorators.scala

```diff
@@ -112,9 +112,15 @@ object Decorators {
         else x1 :: xs1
       }
 
-    def foldRightBN[U](z: => U)(op: (T, => U) => U): U = xs match {
-      case Nil => z
-      case x :: xs1 => op(x, xs1.foldRightBN(z)(op))
+    def foldRightBN[U](z: => U)(op: (T, => U) => U): U =
+      xs.reverse.foldLeftBN(z)(op)
+
+    final def foldLeftBN[U](acc: => U)(op: (T, => U) => U): U = {
+      @tailrec def fold(xs: List[T], acc: => U): U = xs match {
+        case x :: xs1 => fold(xs1, op(x, acc))
+        case Nil => acc
+      }
+      fold(xs, acc)
     }
```

Comment:
Should be
```scala
final def foldLeftBN[U](acc: => U)(op: (=> U, T) => U): U = {
  ...
  case x :: xs1 => fold(xs1, op(acc, x))
```
That's what I meant by "accumulator is the first arg of the closure"

Reply:
That is right, but if I do that the `op` must be put in another closure to reverse the parameter order.
I will make `fold` an inner method of `foldRightBN` and avoid this altogether.